### PR TITLE
Add configuration property for RequestStats

### PR DESCRIPTION
**Pull request by viczhang861**
Provide an option to optimize jetty thread pool usage.  Excessive usage of jetty threads to update RequestStats object was observed when workload is high.

### CODE REVIEW

File: http-server/src/main/java/com/facebook/airlift/http/server/StatsRecordingHandler.java [hunk 1]
```diff
@@ -23,21 +23,31 @@
 import java.util.concurrent.TimeUnit;
 
 import static java.lang.Math.max;
+import static java.util.Objects.requireNonNull;
 
 public class StatsRecordingHandler
         implements RequestLog
 {
     private final RequestStats stats;
+    private final boolean statsRecordEnabled;
```
Comment (wenleix) [inline http-server/src/main/java/com/facebook/airlift/http/server/StatsRecordingHandler.java]: When this boolean is true, `RequestStats` is provided but not used. This is a bit weird.

Another way would just to make `stats` field `Optional`.

**Comment by wenleix** [comment]
@viczhang861 : Thanks ! Do you mind explain a bit about the motivation? (e.g. we observed excessive locking due to log)

File: http-server/src/main/java/com/facebook/airlift/http/server/StatsRecordingHandler.java (hunk 1 above)
Comment (wenleix) [reply http-server/src/main/java/com/facebook/airlift/http/server/StatsRecordingHandler.java]: I am thinking to just have 1 field (`Optional<RequestStats> stats`). So if `stats` is not present, is viewed as stats record is disabled

Comment (viczhang861) [reply http-server/src/main/java/com/facebook/airlift/http/server/StatsRecordingHandler.java]: Good point, updated.

File: http-server/src/main/java/com/facebook/airlift/http/server/HttpServer.java [hunk 2]
```diff
@@ -358,7 +358,7 @@ public HttpServer(HttpServerInfo httpServerInfo,
         handlers.addHandler(createServletContext(theServlet, parameters, filters, tokenManager, loginService, "http", "https"));
 
         RequestLogHandler statsRecorder = new RequestLogHandler();
```
Comment (arhimondr) [inline http-server/src/main/java/com/facebook/airlift/http/server/HttpServer.java]: How about:

```
if(config.isRequestStatsEnabled()){
  RequestLogHandler statsRecorder = new RequestLogHandler();
  statsRecorder.setRequestLog(new StatsRecordingHandler(stats));
  handlers.addHandler(statsRecorder);
}
```

?

Comment (viczhang861) [reply http-server/src/main/java/com/facebook/airlift/http/server/HttpServer.java]: This is cleaner, I didn't know this handler is optional.  cc @wenleix

**Comment by wenleix** [comment]
Merged #11, thanks for the contribution!

**Pull request merged**